comp/aggregator/demultiplexer/demultiplexer.go

```diff
@@ -23,17 +25,31 @@ type dependencies struct {
 	Params Params
 }
 
-func newDemultiplexer(deps dependencies) (Component, error) {
+type demultiplexer struct {
+	*aggregator.AgentDemultiplexer
+}
+
+func newDemultiplexer(deps dependencies) (Component, diagnosesendermanager.Component, error) {
 	hostnameDetected, err := hostname.Get(context.TODO())
 	if err != nil {
-		return nil, deps.Log.Errorf("Error while getting hostname, exiting: %v", err)
+		return nil, nil, deps.Log.Errorf("Error while getting hostname, exiting: %v", err)
 	}
 
-	demux := aggregator.InitAndStartAgentDemultiplexer(
+	agentDemultiplexer := aggregator.InitAndStartAgentDemultiplexer(
 		deps.Log,
 		deps.SharedForwarder,
 		deps.Params.Options,
 		hostnameDetected)
+	demultiplexer := demultiplexer{
+		AgentDemultiplexer: agentDemultiplexer,
+	}
+
+	// Implement also diagnosesendermanager.Component to make sure
+	// either demultiplexer.Component nor diagnosesendermanager.Component is created.
+	return demultiplexer, demultiplexer, nil
```

Comment:
Not sure I understand the comment. Why are we returning twice the same type ?

Reply:
I have updated the comment. Let me know if it is clear enough.